bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/parallacticangle/ParallacticAngleSupportInst.java

```diff
@@ -28,28 +26,10 @@ protected ParallacticAngleSupportInst(SPComponentType type) {
      * Instruments who need specific modifications based on settings should override this, e.g. GMOS.
      */
     @Override
-    public Option<Angle> calculateParallacticAngle(ISPObservation obs) {
-        final Option<TargetCalculator> targetCalculatorOption = ObsTargetCalculatorService.targetCalculationForJava(obs);
-        if (!targetCalculatorOption.isEmpty()) {
-            final TargetCalculator targetCalculator = targetCalculatorOption.getValue();
-
-            // Calculate the weighted angle.
-            scala.Option<Object> angleOption = targetCalculator.weightedMeanParallacticAngle();
-            if (angleOption.nonEmpty()) {
-                // Calculate the parallactic angle.
-                final double dAngle = (Double) angleOption.get();
-                final edu.gemini.skycalc.Angle angle = (new edu.gemini.skycalc.Angle(dAngle, edu.gemini.skycalc.Angle.Unit.DEGREES)).toPositive();
-                return new Some<>(angle);
-            } else return None.instance();
-        } else return None.instance();
-    }
-
-    /**
-     * By default, assume that the instrument is compatible with the parallactic angle feature unless indicated
-     * otherwise.
-     */
-    @Override
-    public boolean isCompatibleWithMeanParallacticAngleMode() {
-        return true;
+    public Option<Angle> calculateParallacticAngle(final ISPObservation obs) {
+        return ImOption.fromScalaOpt(ObsTargetCalculatorService.targetCalculation(obs))
+                .flatMap(targetCalculator -> ImOption.fromScalaOpt(targetCalculator.weightedMeanParallacticAngle()))
+                .map(angleObj -> (double) angleObj)
+                .map(angle -> (new edu.gemini.skycalc.Angle(angle, edu.gemini.skycalc.Angle.Unit.DEGREES)).toPositive());
```

Comment:
I suppose we should get in the habit of switching things over to that.
I made the changes and everything appears to be working well!

If someone could 👍  this this morning or offer more feedback, that would be excellent so I can move ahead with the `Swing.EDT` issues for the `PositionAnglePanel` and `ParallacticAngleControls` so that we can attempt to get a test release out today.